packages/core/saved-objects/core-saved-objects-utils-server/src/saved_objects_test_utils.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isEqual } from 'lodash';
+
+/**
+ * Determines if a given Set is equal to another given Set. Set types must be the same, and comparable.
+ *
+ * @param setA the first Set to compare
+ * @param setB the second Set to compare
+ * @returns {boolean} True if Set A is equal to Set B
+ */
+export function setsAreEqual<T>(setA: Set<T>, setB: Set<T>) {
+  // console.log(`*** SET A: ${Array.from(setA)}`);
+  // console.log(`*** SET B: ${Array.from(setB)}`);
+  return isEqual(Array.from(setA).sort(), Array.from(setB).sort());
```

Comment:
Beware: converting a Set to an Array for comparison kills multiple kittens every time the function is called. (poorly optimized - can just check same length and presence of each entry from A in B, would avoid sorting two arrays to check equality on a Set 😅 ).

Reply:
I reimplemented this with no conversions, using foreEach and a dummy exception to break the loop early when inequality is detected. Hopefully no more kittens will be destroyed 🙏 

```
  if (setA.size !== setB.size) return false;

  var NotEqualException = {};
  try {
    setA.forEach(element => {
      // End the loop early if we found an inequality
      if(!setB.has(element)) throw NotEqualException;
    });
  } catch (NotEqualException) {
    return false;
  }

  return true;
```

Reply:
I might be missing something, but cannot we just do something like this (instead of `try/catch` block)?
```ts
  for (const element of setA) {
    if (!setB.has(element)) {
      return false;
    }
  }
```

Reply:
You're not missing anything. I just overcomplicated it due to thinking I was limited to the forEach with Sets. Thanks!